### DISCUSSION TITLE
Transit compartment in MoBi

### DISCUSSION THREAD

**Discussion by Sulafa93** [Open-Systems-Pharmacology/Forum#1880] [Questions & Problems]
Hi all! 

I am working on building a model in MoBi for a long-acting intramuscular injectable. The drug profile exhibits a double-peak pattern, indicating an initial fast release followed by a controlled release phase. Do you have any suggestions or best practices for modeling this type of release profile in MoBi? Any guidance on setting up a transit compartment or alternative approaches would be greatly appreciated.

Thanks!

**Comment by tobiasK2001** [comment]
Hi, here are my two pence: an alternative approach you could use to approach this empirical is to split the dose in two parts (slow and fast release) and use two Weibull functions to describe the fast and slow release. Similar like Nina Hnke did it for the Rosuvastatin model (https://github.com/Open-Systems-Pharmacology/Rosuvastatin-Model/blob/master/Rosuvastatin-Model.pksim5). Easiest would be to start of in PK-sim with user defined admin protocol like this 
![image](https://github.com/user-attachments/assets/cb519b42-5d6c-435d-8a3d-1bbf25022e11).
Than you can modify the advanced protocol in Mobi with the user defined part from PK-sim 
![image](https://github.com/user-attachments/assets/e7703941-7a70-4663-9fd1-89258e46507f)
and connect a formulation to it:
![image](https://github.com/user-attachments/assets/2066d5a3-1ca1-4997-b4d1-6a27f379a13b)

Hope that might help!

Best, Tobias

**Comment by Sulafa93** [comment]
Hi Tobias,
Thanks for your reply! I considered splitting the dose in PK-Sim and using the advanced protocol, but it seems like the Weibull function is only available for oral formulations (I am modeling an IM injection). Thanks again!

**Comment by StephanSchaller** [comment]
Just to have others also see a reply we provided offline:

> MoBi is flexible enough and it should be reasonably simple to add any type of empirical dissolutions or release function you fancy for your model.
We have outlined this for a double-Weibull on GitHub: https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/1701

Reach out if you need more info to get this implemented (we also hope to have this natively in PK-Sim for an upcoming release (likely V13)).

**Comment by Sulafa93** [comment]
Hello all,

Thank you for suggesting the double-Weibull function. 

Here is the workflow I followed:

1. In PK-Sim, I created a simple administration protocol using the user-defined administration type, where the entire dose was administered into the muscle interstitial space.
2. I defined a first-order formulation using the drug half-life.
3. After running the simulation, I exported it to MoBi, where I modified the event by incorporating the formula of the double Weibull function, with Lag Time 2 affecting the appearance of the second peak.
![Picture1](https://github.com/user-attachments/assets/feeb42ad-0b65-46f7-8e05-0ef50fd841a5)

Here is the simulation results: 
![Picture2](https://github.com/user-attachments/assets/67b87091-8824-483f-a45e-d9a10cae8258)

The results indicate that whatever is released gets rapidly absorbed, but I don’t believe that reflects reality (especially for the second fraction). I suspect that I may need to introduce a depot compartment, reduce the absorption rate into systemic circulation, or adjust the muscle Kp scaler. It is also worth mentioning that there are no in vitro dissolution data available to describe the release, and the alpha1,2 and beta1,2 values were predicted.

Would you recommend any specific approach to better capture the expected absorption behavior? Are there other parameters I should consider adjusting?

Thanks in advance for your insights!
